### PR TITLE
[Enhancement](thread) FE fetch_data use brpc thread locally in BE 

### PR DESCRIPTION
**Pull request by zclllyybb**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

1. do `fetch_data` in brpc thread locally
2. add more gauge for fragment async thread
3. fix wrong log info for FifoThreadPool

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by zclllyybb** [comment]
run buildall

File: be/src/service/internal_service.cpp [hunk 1]
```diff
@@ -665,11 +665,12 @@ void PInternalService::cancel_plan_fragment(google::protobuf::RpcController* /*c
...
-        _exec_env->result_mgr()->fetch_data(request->finst_id(), ctx);
-    });
+    bool ret = ExecEnv::GetInstance()->fragment_mgr()->get_thread_pool()->submit_func(
+            [this, controller, request, result, done]() {
+                auto* cntl = static_cast<brpc::Controller*>(controller);
+                auto* ctx = new GetResultBatchCtx(cntl, result, done);
+                _exec_env->result_mgr()->fetch_data(request->finst_id(), ctx);
+            });
     if (!ret) {
         offer_failed(result, done, _heavy_work_pool);
```
Comment (HappenLee) [inline be/src/service/internal_service.cpp:675]: not add the pool should not call `offer_failed` with `_heavy_work_pool`

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**